src/include/txn_inline.h

```diff
@@ -1005,6 +1017,17 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
     /* If there's no visible update in the update chain or ondisk, check the history store file. */
     if (F_ISSET(S2C(session), WT_CONN_HS_OPEN) && !F_ISSET(session->dhandle, WT_DHANDLE_HS)) {
         __wt_timing_stress(session, WT_TIMING_STRESS_HS_SEARCH);
+
+        /*
+         * Use the restored update data as base value if it is restored as part of RTS instead of
+         * existing on-disk value.
+         */
+        if (restored_upd != NULL) {
```

Comment:
I don't understand why we can't use the on disk value here. Can you elaborate the scenario?

Reply:
A scenario where the on-disk version is aborted by the RTS by restoring an update from the history store. But the page is not yet evicted to reload the new image that was checkpointed as part of RTS. During this time if a search operation that wants to read a history store data will use the on-disk aborted update as a base value to return any history store modify updates.

Reply:
This is not nice. Can we force the page to evict so the following search can use the onpage value?

Reply:
Hmm, then we have to evict all the pages that are fixed by the RTS during recovery or application initiated flows that will unnecessary adds time to RTS. Do see any problem that can happen now/in future in using the restored update as a base value?

Reply:
I don't see any problems logically but it is something very tricky to maintain.

Reply:
Ok. Let me know if you find an alternative way to solve this problem. I will also check any simpler method other than eviction.